python_modules/dagster/dagster/_core/definitions/asset_graph_differ.py

```diff
@@ -162,10 +229,90 @@ def _compare_base_and_branch_assets(
 
         return changes
 
-    def get_changes_for_asset(self, asset_key: "AssetKey") -> Sequence[AssetDefinitionChangeType]:
+    def _compare_base_and_branch_assets_detail(
```

Comment:
Might be worth trying to unify the impl of this and the base asset definition change logic if possible just to avoid repeated work and to ensure the behaviors continue to line up if we change one/both in the future.

One way I could see doing this is making a helper fn that just returns sequence of `(ChangeType, Optional[ChangeDetail])` and which has a flag whether to do the work to populate the latter. Then you have two public APIs that rely on the same helper.